controllers/cluster_controller.go

```diff
@@ -50,6 +52,38 @@ const (
 	deleteRequeueAfter = 5 * time.Second
 )
 
+var (
+	clusterControlPlaneReady = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "capi_cluster_control_plane_ready",
+			Help: "Cluster control plane is ready if set to 1 and not if 0.",
+		},
+		[]string{"cluster", "namespace"},
+	)
+	clusterInfrastructureReady = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "capi_cluster_infrastructure_ready",
+			Help: "Cluster infrastructure is ready if set to 1 and not if 0.",
+		},
+		[]string{"cluster", "namespace"},
+	)
+	clusterKubeconfigReady = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "capi_cluster_kubeconfig_ready",
+			Help: "Cluster kubeconfig is ready if set to 1 and not if 0.",
+		},
+		[]string{"cluster", "namespace"},
+	)
+)
+
+func init() {
+	metrics.Registry.MustRegister(
+		clusterControlPlaneReady,
+		clusterInfrastructureReady,
+		clusterKubeconfigReady,
+	)
+}
```

Comment:
Does this need to be done with init? Could we instead initialize this in `SetupWithManager` instead?

Reply:
I was thinking of the same thing as well. All the conventions I've see when initializing these metrics were in `init` but I don't see any problem with putting them in `SetupWithManager`.

Reply:
@detiber Actually, the one difference is that it makes it hard to test the metrics if they are registered within `SetupWithManager`.

Reply:
Thanks for the context, no objections from me, then.